runtime/doc/lua.txt

```diff
@@ -3388,6 +3405,29 @@ Iter:find({f})                                                   *Iter:find()*
     Return: ~
         any
 
+Iter:flatten({depth})                                         *Iter:flatten()*
```

Comment:
`flat` is shorter and is what https://github.com/neovim/neovim/issues/26454 specifies.
```suggestion
Iter:flat({depth})                                         *Iter:flat()*
```

Reply:
See https://github.com/neovim/neovim/pull/26786#discussion_r1438688152

Reply:
Shorter is better. @gpanders do you mind if we use `flat`

Reply:
> Shorter is better. @gpanders do you mind if we use `flat`

Consistent is also better. `flat()` feels like a very bizarre function name, it is neither a noun nor a verb. The rest of the API in vim.iter() (and our API in general) uses verbs for function names to indicate it “does something”.

`flatten` is also, as far as I know, much more common in other languages. In fact, aside from the one JavaScript implementation you linked to I’ve never seen such a function called `flat` https://rosettacode.org/wiki/Flatten_a_list